Studio: Show sync selection size (e.g. 1.2 GB / 2 GB) in site push dialog

katinthehatsite opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "Fixes" keyword and use "Related to" instead.
-->

## Related issues

Closes STU-908

## Proposed Changes

This PR adds the bar that displays the size of the Studio site when the user is trying to perform the `Push` operation:

<img width="725" height="456" alt="Screenshot 2025-11-25 at 3 26 54 PM" src="https://github.com/user-attachments/assets/0bb18dbb-f6c4-4891-b496-a857924c816e" />

If the size is over the limit, the user is seeing the following warning:

<img width="817" height="481" alt="Screenshot 2025-11-25 at 3 27 32 PM" src="https://github.com/user-attachments/assets/e8c14204-d9df-4878-9a60-fe319122413c" />

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Pull the changes from this branch
* Ensure that you have a Studio site that is over 2GB (you can run `cd wp-content/uploads` and 
`dd if=/dev/zero of=dummyfile.dat bs=1M count=2048` to inflate the site size if needed)
* Navigate to the `Sync` tab
* Make sure that you have at least one site connected
* Click on the `Push` option to open the sync dialog
* Observe that the progress bar shows as empty (0 bytes)
* Next, start selecting things and confirm that the progress bar starts to fill in
* Select everything that confirm that you can see over the limit warning and the progress bar

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Have you checked for TypeScript, React or other console errors?

Commit:
Add basic implementation

Commit:
Account for footer sizing

Commit:
Add custom progress bar

Commit:
Minor styling

Commit:
Final adjustments for the style

katinthehatsite commented on src/components/progress-bar.tsx:
```diff
@@ -60,3 +60,57 @@ export function ProgressBarWithAutoIncrement( {
+	trackColor?: string;
+};
+
+export function TwoColorProgressBar( {
```
I tried to achieve this with `ProgressBar` but that component is too simple so I opted for this custom simple implementation.

katinthehatsite commented on src/modules/sync/components/sync-dialog.tsx:
```diff
@@ -218,13 +222,23 @@ export function SyncDialog( {
 
+	const getBottomPadding = () => {
+		if ( type === 'pull' ) {
+			return 'pb-[70px]'; // Original padding for pull
```
I left these comments for the ease of reading the function. They could be removed if they seem self-explanatory

Commit:
Component refactor

Commit:
Ensure we use color variables

katinthehatsite commented on src/modules/sync/components/sync-dialog.tsx:
```diff
@@ -321,7 +335,18 @@ export function SyncDialog( {
+						</div>
+					) }
 					{ type === 'push' && isPushSelectionOverLimit && (
 						<Notice status="warning" isDismissible={ false } className="mb-4">
```
I left the notice as warning even though the designs had it in blue. I tried the `info` version and this is what it looked like:

<img width="667" height="137" alt="Image" src="https://github.com/user-attachments/assets/8e909b22-64ee-4867-98c5-59e0dd3373f8" />

To me it looked a bit too pale to be well noticed but we can change it if that's the preference.

wojtekn commented:
> This PR adds the bar that displays the size of the Studio site when the user is trying to perform the Push operation:

@katinthehatsite the bar looks great. What if we added 2 GB on the right and changed 0 bytes to 0 B, to make it consistent with elements used in the Hosting Dashboard?

Commit:
Rename bytes

Commit:
Add total limit on the right

katinthehatsite commented:
> @katinthehatsite the bar looks great. What if we added 2 GB on the right and changed 0 bytes to 0 B, to make it consistent with elements used in the Hosting Dashboard?

@wojtekn this should be taken care of, let me know what you think

wojtekn commented:
@katinthehatsite I haven't reviewed the code, but the UI looks great.

I just noticed one bug, as it looks it takes into the account everything even when I select one file?

<img width="555" height="507" alt="Screenshot 2025-11-27 at 15 10 17" src="https://github.com/user-attachments/assets/07199543-889d-4d88-afd5-bea24a488944" />

<img width="553" height="501" alt="Screenshot 2025-11-27 at 15 09 03" src="https://github.com/user-attachments/assets/9de79659-594f-4de4-a388-0c70ca5e5de3" />

<img width="630" height="118" alt="Screenshot 2025-11-27 at 15 09 07" src="https://github.com/user-attachments/assets/51d65c32-4024-4e5d-9947-f783dbab55bf" />

sejas committed:
Merge branch 'trunk' of github.com:Automattic/studio into add/site-size-display-for-sync

sejas reviewed (commented):
Works pretty good. Thanks for adding the progress bar showing the size of file selection.
I found the issue mentioned by Wojtek. The calculation only goes two levels deep, so for uploads it shows the total of the folder. I'll commit the fix to this branch.

sejas committed:
Make the file size calculation recursive

sejas committed:
Add special condition for mu-plugins

sejas committed:
For mu-plugins calculate it only for selected nodes because we have sqlite mu-plugin in the file-system

sejas committed:
Reuse SYNC_PUSH_SIZE_LIMIT_BYTES constant

sejas reviewed (approved):
I confirm it shows the size of a selected file with `B` bytes.
<img width="1084" height="773" alt="Image" src="https://github.com/user-attachments/assets/fc5c91e1-e291-45ab-a60e-b5c119418d01" />

It also counts the individual elements.

<img width="1084" height="773" alt="Image" src="https://github.com/user-attachments/assets/7c1c8894-c34c-4614-8990-7daf43a208f4" />

And it displays the warning when the selection is over the limit.

<img width="1084" height="773" alt="Image" src="https://github.com/user-attachments/assets/186ac39b-cf90-429f-a37a-705d1d4ec966" />

sejas committed:
Fix tests by making wpContentIndeterminate more real

katinthehatsite commented:
Nice, thanks for making adjustments and bringing this PR over the finish line ❤️
